FrontEnd/src/BookingPage/BookingPage.jsx

```diff
@@ -0,0 +1,126 @@
+import "./BookingPage.css";
+import { useLocation } from "react-router-dom";
+import { useState } from "react";
+import { requestAppointment } from "./BookingDB";
+import Modal from "../Modal";
+import NotificationsPage from "../Notifications/NotificationsPage";
+
+function BookingForm({ userData }) {
+  const location = useLocation();
+  const { post, userUID } = location.state || {};
+  const invisibleComponent = false;
```

Comment:
What is this `invisibleComponent` ?

Reply:
Passing the "vendorID" data with useContext created a bug in the application so I opted for using props to pass the data down but I did not want the component to show